source/gfoidl.Base64/Internal/Base64UrlEncoder/Base64UrlEncoder.Encode.cs

```diff
@@ -231,8 +231,32 @@ private static int GetNumBase64PaddingCharsAddedByEncode(int dataLength)
             // 0 -> 0
             // 1 -> 2
             // 2 -> 1
+            int mod3 = FastMod3(dataLength);
+            return mod3 == 0 ? 0 : (3 - mod3);
+        }
+        //---------------------------------------------------------------------
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int FastMod3(int value)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                // We use modified Daniel Lemire's fastmod algorithm (https://github.com/dotnet/runtime/pull/406),
+                // which allows to avoid the long multiplication if the divisor is less than 2**31.
+                Debug.Assert(value <= int.MaxValue);
```

Comment:
Looks to be useless... 
A better assertion would be value >= 0, but in this case the tests will fail in DEBUG.

Should we thow an exception here ? It will add a new branch...
Should we ignore the assertion ? The exception will be thrown by `GetBase64EncodedLength()` anyway.

Reply:
OK you reply just before to my questions!

Reply:
Final question:
Is the assertion stlll useful ? Or should I switch back to uint value parameter?

Reply:
The test with `value = -1` causes the assert. You've changed the argument to `int`, so the condition will be true and the assert won't fire. Keep the `uint`, as negative values casted to `uint` will be greater than `int.MaxValue` and the assert will trigger. I'd like to keep the assert -- even if not strictly necessary as you pointed out -- but if I or anyone else copies the code one could get trapped easily.

See https://github.com/gfoidl/Base64/pull/114#issuecomment-587568326 for a fix which throws before the assert will be reached.